core/src/main/scala/kafka/MetadataLogConfig.scala

```diff
@@ -32,13 +32,13 @@ final case class MetadataLogConfig(
 )
 
 object MetadataLogConfig {
-  def apply(config: AbstractConfig, maxBatchSizeInBytes: Int, maxFetchSizeInBytes: Int): MetadataLogConfig = {
+  def apply(config: KafkaConfig, maxBatchSizeInBytes: Int, maxFetchSizeInBytes: Int): MetadataLogConfig = {
     new MetadataLogConfig(
-      config.getInt(KRaftConfigs.METADATA_LOG_SEGMENT_BYTES_CONFIG),
-      config.getInt(KRaftConfigs.METADATA_LOG_SEGMENT_MIN_BYTES_CONFIG),
-      config.getLong(KRaftConfigs.METADATA_LOG_SEGMENT_MILLIS_CONFIG),
-      config.getLong(KRaftConfigs.METADATA_MAX_RETENTION_BYTES_CONFIG),
-      config.getLong(KRaftConfigs.METADATA_MAX_RETENTION_MILLIS_CONFIG),
+      config.metadataLogSegmentBytes,
+      config.metadataLogSegmentMinBytes,
+      config.metadataLogSegmentMillis,
+      config.metadataRetentionBytes,
+      config.metadataRetentionMillis,
       maxBatchSizeInBytes,
       maxFetchSizeInBytes,
       ServerLogConfigs.LOG_DELETE_DELAY_MS_DEFAULT,
```

Comment:
Could you change line#45 to use `config.nodeId`?

Reply:
thanks for the comment, nice catch!
let me update that